src/structures/TeamMember.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const Base = require('./Base');
+const { MembershipStates } = require('../util/Constants');
+
+/**
+ * Represents a Client OAuth2 Application Team Member.
+ * @extends {Base}
+ */
+class TeamMember extends Base {
+  constructor(client, team, data) {
+    super(client);
+
+    /**
+     * The Team this member is part of
+     * @type {Team}
+     */
+    this.team = team;
+
+    this._patch(data);
+  }
+
+  _patch(data) {
+    /**
+     * The permissions this Team Member has with reguard to the team
+     * @type {string[]}
+     */
+    this.permissions = data.permissions;
+
+    /**
+     * The permissions this Team Member has with reguard to the team
+     * @type {MembershipStates}
+     */
+    this.membershipState = MembershipStates[data.membership_state];
+
+    /**
+     * The user for this Team Member
+     * @type {User}
+     */
+    this.user = this.client.users.add(data.user);
+
+    /**
+     * The ID of the Team Member
```

Comment:
I meant it. This is redundant...

Reply:
TeamMembers are keyed by id in Team#members, the only id we have to relate to is the user id, which is no different than how GuildMembers' ids work.

I thought your original complaint is we weren't using the User instead of the raw user data, so that's why I made the change I did, which really doesn't matter which way.